### PR TITLE
Updating build mechanisms to work for tagging.

### PR DESCRIPTION
**Pull request by swernli**
@jstarks 
This change introduces a new approach for versioning the project:

 - The build version in the project.json file is the official central version, and must be of the form Major.Minor.Revision. No prerelease additions (ie: "-alpha") supported since PowerShell doesn't support them.

 - All other versions (.nuspec package metadata, .psd1 module metadata) remain at 0.0.0, which represents the version for local developer builds.

 - At AppVeyor build time, the project.json version is written into the .nuspec and .psd1 files.  If this is a non-tagged build, it is a developer feed build and will include a build number in the version (dynamically added to the project.json as well), making the full version format Major.Minor.Revision.Build.  The build number will not be included for tagged release builds.

### CODE REVIEW

**Comment by swernli** [comment]
@jstarks I think we are finally ready for this!

File: appveyor.yml [hunk 1]
```diff
@@ -19,31 +19,38 @@ build_script:
...
+      $project.version = $version
+      ConvertTo-Json $project -Depth 100 | Out-File -Encoding UTF8 $projectFile
+    }
+
+    # Replace module manifest version.
     $manifest = "src\Docker.PowerShell\Docker.psd1"
-    (Get-Content $manifest -Raw) -replace "ModuleVersion.+","ModuleVersion = '$psversion'" | Out-File $manifest
+    (Get-Content $manifest -Raw) -replace "ModuleVersion.+","ModuleVersion = '$version'" | Out-File $manifest
+
+    # Replace nuspec version.
```
Comment (jstarks) [inline appveyor.yml]: Why is this necessary? Doesn't the nuget pack line do this already?
